### PR TITLE
verify bucket names in wizard

### LINKED ISSUE #1814: BlockingDataflowPipelineRunner expected a valid 'gs://' path but was given 'gs://SET_YOUR_BUCKET_NAME_HERE/AND_STAGING_DIRECTORY'
When attempting to run example pipeline

```
Exception in thread "main" java.lang.RuntimeException: Failed to construct instance from factory method BlockingDataflowPipelineRunner#fromOptions(interface com.google.cloud.dataflow.sdk.options.PipelineOptions)
	at com.google.cloud.dataflow.sdk.util.InstanceBuilder.buildFromMethod(InstanceBuilder.java:234)
	at com.google.cloud.dataflow.sdk.util.InstanceBuilder.build(InstanceBuilder.java:163)
	at com.google.cloud.dataflow.sdk.runners.PipelineRunner.fromOptions(PipelineRunner.java:58)
	at com.google.cloud.dataflow.sdk.Pipeline.create(Pipeline.java:135)
	at com.google.cloud.dataflow.examples.MinimalWordCount.main(MinimalWordCount.java:72)
Caused by: java.lang.reflect.InvocationTargetException
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:606)
	at com.google.cloud.dataflow.sdk.util.InstanceBuilder.buildFromMethod(InstanceBuilder.java:223)
	... 4 more
Caused by: java.lang.IllegalArgumentException: BlockingDataflowPipelineRunner expected a valid 'gs://' path but was given 'gs://SET_YOUR_BUCKET_NAME_HERE/AND_STAGING_DIRECTORY'
	at com.google.cloud.dataflow.sdk.util.DataflowPathValidator.getGcsPath(DataflowPathValidator.java:93)
	at com.google.cloud.dataflow.sdk.util.DataflowPathValidator.verifyPath(DataflowPathValidator.java:69)
	at com.google.cloud.dataflow.sdk.util.DataflowPathValidator.validateOutputFilePrefixSupported(DataflowPathValidator.java:62)
	at com.google.cloud.dataflow.sdk.runners.DataflowPipelineRunner.fromOptions(DataflowPipelineRunner.java:274)
	at com.google.cloud.dataflow.sdk.runners.BlockingDataflowPipelineRunner.fromOptions(BlockingDataflowPipelineRunner.java:82)
	... 9 more
Caused by: java.lang.IllegalArgumentException: GCS bucket names must contain only lowercase letters, numbers, dashes (-), underscores (_), and dots (.). Bucket names must start and end with a number or letter. See https://developers.google.com/storage/docs/bucketnaming for more details.  Bucket name: SET_YOUR_BUCKET_NAME_HERE
	at com.google.cloud.dataflow.sdk.repackaged.com.google.common.base.Preconditions.checkArgument(Preconditions.java:122)
	at com.google.cloud.dataflow.sdk.util.gcsfs.GcsPath.<init>(GcsPath.java:207)
	at com.google.cloud.dataflow.sdk.util.gcsfs.GcsPath.fromUri(GcsPath.java:121)
	at com.google.cloud.dataflow.sdk.util.DataflowPathValidator.getGcsPath(DataflowPathValidator.java:91)
	... 13 more
```

### PR DESCRIPTION
**Pull request by elharo**
@tgroh  fix #1814

### CODE REVIEW

File: plugins/com.google.cloud.tools.eclipse.dataflow.ui.test/src/com/google/cloud/tools/eclipse/dataflow/ui/preferences/RunOptionsDefaultsComponentTest.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.runners.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class RunOptionsDefaultsComponentTest {
+  
+  private @Mock DataflowPreferences preferences;
+  private Shell shell;
+  private MessageTarget messageTarget = Mockito.mock(MessageTarget.class);
```
Comment (tgroh) [inline plugins/com.google.cloud.tools.eclipse.dataflow.ui.test/src/com/google/cloud/tools/eclipse/dataflow/ui/preferences/RunOptionsDefaultsComponentTest.java]: Use an annotation

File: plugins/com.google.cloud.tools.eclipse.dataflow.ui.test/src/com/google/cloud/tools/eclipse/dataflow/ui/preferences/RunOptionsDefaultsComponentTest.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.runners.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class RunOptionsDefaultsComponentTest {
+  
+  private @Mock DataflowPreferences preferences;
```
Comment (tgroh) [inline plugins/com.google.cloud.tools.eclipse.dataflow.ui.test/src/com/google/cloud/tools/eclipse/dataflow/ui/preferences/RunOptionsDefaultsComponentTest.java]: Is the interleaved annotation standard here?

File: plugins/com.google.cloud.tools.eclipse.dataflow.ui.test/src/com/google/cloud/tools/eclipse/dataflow/ui/preferences/RunOptionsDefaultsComponentTest.java [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+  private Shell shell;
+  private MessageTarget messageTarget = Mockito.mock(MessageTarget.class);
+  private RunOptionsDefaultsComponent component;
+  private Composite composite;
+
+  @Before
+  public void setUp() {
+    shell = new Shell(Display.getCurrent());
```
Comment (tgroh) [inline plugins/com.google.cloud.tools.eclipse.dataflow.ui.test/src/com/google/cloud/tools/eclipse/dataflow/ui/preferences/RunOptionsDefaultsComponentTest.java]: You should initialize your mocks here

File: plugins/com.google.cloud.tools.eclipse.dataflow.ui/src/com/google/cloud/tools/eclipse/dataflow/ui/preferences/RunOptionsDefaultsComponent.java [hunk 4]
```diff
@@ -141,6 +144,16 @@ public void onComplete(String contents) {
     messageTarget.setInfo("Set Pipeline Run Option Defaults");
   }
 
+  public RunOptionsDefaultsComponent(
+      Composite composite,
+      int numColumns,
+      DialogPageMessageTarget dialogPageMessageTarget,
+      DataflowPreferences prefs,
+      WizardPage page) {
+    this(composite, numColumns, dialogPageMessageTarget, prefs);
```
Comment (tgroh) [inline plugins/com.google.cloud.tools.eclipse.dataflow.ui/src/com/google/cloud/tools/eclipse/dataflow/ui/preferences/RunOptionsDefaultsComponent.java]: If page is nullable (which it appears it is), I'd prefer to have all of the implementation in this constructor, and have the simpler constructor just pass in null for the `WizardPage` param.

File: plugins/com.google.cloud.tools.eclipse.dataflow.ui/src/com/google/cloud/tools/eclipse/dataflow/ui/preferences/RunOptionsDefaultsComponent.java [hunk 5]
```diff
@@ -259,7 +279,7 @@ public void focusLost(FocusEvent event) {
       updateStagingLocations(getProject());
     }
   }
-
+
```
Comment (tgroh) [inline plugins/com.google.cloud.tools.eclipse.dataflow.ui/src/com/google/cloud/tools/eclipse/dataflow/ui/preferences/RunOptionsDefaultsComponent.java]: ?

File: plugins/com.google.cloud.tools.eclipse.dataflow.ui.test/src/com/google/cloud/tools/eclipse/dataflow/ui/preferences/RunOptionsDefaultsComponentTest.java [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+    composite = new Composite(shell, SWT.NONE);
+    component = new RunOptionsDefaultsComponent(composite, 3, messageTarget, preferences);
+  }
+
+  @After
+  public void tearDown() {
+    if (shell != null && !shell.isDisposed()) {
```
Comment (chanseokoh) [inline plugins/com.google.cloud.tools.eclipse.dataflow.ui.test/src/com/google/cloud/tools/eclipse/dataflow/ui/preferences/RunOptionsDefaultsComponentTest.java]: `shell` is never null (always created in `setUp()`), and it is okay to call `dispose()` multiple times, so you don't need this if-stmt.

**Comment by chanseokoh** [comment]
The simplest solution for this is to flip `<useUIThread>` to true in `pom.xml` in the test bundle.

```
tesGetControl(com.google.cloud.tools.eclipse.dataflow.ui.preferences.RunOptionsDefaultsComponentTest)  Time elapsed: 0.002 sec  <<< ERROR!
org.eclipse.swt.SWTException: Invalid thread access
	at org.eclipse.swt.SWT.error(SWT.java:4491)
```

BTW, the above method name is `tesG...`.

**Comment by tgroh** [comment]
However, failing tests eg:
testCloudProjectText(com.google.cloud.tools.eclipse.dataflow.ui.preferences.RunOptionsDefaultsComponentTest)  Time elapsed: 0.003 sec  <<< ERROR!
org.eclipse.swt.SWTException: Invalid thread access
	at org.eclipse.swt.SWT.error(SWT.java:4491)

**Pull request merged**